src/main/java/de/komoot/photon/opensearch/OpenSearchSearchHandler.java

```diff
@@ -34,7 +34,11 @@ public List<PhotonResult> search(SimpleSearchRequest request) {
 
         List<PhotonResult> ret = new ArrayList<>();
         for (var hit : results.hits().hits()) {
-            ret.add(hit.source().setScore(hit.score()));
+            var score = hit.score();
+            var source = hit.source();
+            if (score != null && source != null) {
+                ret.add(source.setScore(score));
```

Comment:
You need to distinguish another cases here: if source is not null but score is, we'd still want to add the source to the result list.

Reply:
Just don't call setScore() when score is null